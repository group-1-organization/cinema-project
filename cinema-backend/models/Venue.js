const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const venue = new Schema({
    name: {
        type: String,
        minlength: 2,
        required: true
    },
    picture: {
        type: String,
        minlength: 2,
        required: true
    },
    opening: {
        type: String,
        minlength: 2,
        required: true
    },
    description: {
        type: String,
        minlength: 2,
        required: true
    },
    contact: {
        type: String,
        minlength: 2,
        required: true
    },
    offer: {
        type: String,
        minlength: 2,
        required: true
    }
});

module.exports = mongoose.model("venue", venue);