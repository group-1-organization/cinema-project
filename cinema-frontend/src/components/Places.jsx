import Container from "react-bootstrap/Container"
import Row from "react-bootstrap/Row"
import Col from "react-bootstrap/Col"
import Card from "react-bootstrap/Card"
import ListGroup from "react-bootstrap/ListGroup"
import ListGroupItem from "react-bootstrap/ListGroupItem"
import Jumbotron from "react-bootstrap/Jumbotron"
import Button from "react-bootstrap/Button"



const Places = () => {
    return (

        <Container>
            <Jumbotron>
                <center>
                    <h1>Places to Go</h1>
                    <p>
                        A few local restraunts and Services close by, Check them out!!!
                    </p>
                    <p>
                        <Button variant="outline-dark" href="/gettingthere">Getting Here</Button>
                    </p>
                </center>

            </Jumbotron>
            <Row xs={1} lg={3}>
                <Col className="mt-4">
                    <Card className="h-100">
                        <Card.Img variant="top" src="KFC.jpg" style={{ width: "100%" }} />
                        <Card.Body>
                            <Card.Title>KFC</Card.Title>
                            <Card.Text>
                                KFC is an American fast food restaurant chain headquartered in Louisville, Kentucky that specializes in fried chicken
                            </Card.Text>
                        </Card.Body>
                        <ListGroup className="list-group-flush">
                            <ListGroupItem>Opening Times: 9.00 - 10.30</ListGroupItem>
                            <ListGroupItem>Offers: N/A</ListGroupItem>
                            <ListGroupItem>Contact Number: 0344 800 2400</ListGroupItem>
                        </ListGroup>
                    </Card>
                </Col>
                <Col className="mt-4">

                    <Card className="h-100">
                        <Card.Img variant="top" src="5 Guys.jpg" style={{ width: "100%" }} />
                        <Card.Body>
                            <Card.Title>5 Guys</Card.Title>
                            <Card.Text>
                                Five Guys Enterprises LLC is an American fast casual restaurant chain focused on hamburgers, hot dogs, and French fries
                            </Card.Text>
                        </Card.Body>
                        <ListGroup className="list-group-flush">
                            <ListGroupItem>Opening Times: 11.00 - 10.30</ListGroupItem>
                            <ListGroupItem>Offers: 2 for 1 Meals</ListGroupItem>
                            <ListGroupItem>Contact Number: 0344 800 2400</ListGroupItem>
                        </ListGroup>
                    </Card>
                </Col>
                <Col className="mt-4">
                    <Card className="h-100">
                        <Card.Img variant="top" src="FandB.jpg" style={{ width: "100%" }} />
                        <Card.Body>
                            <Card.Title>Frankie and Benny's</Card.Title>
                            <Card.Text>
                                Frankie & Benny's is a chain of Italian-American-themed restaurants in the United Kingdom run by The Restaurant Group.
                            </Card.Text>
                        </Card.Body>
                        <ListGroup className="list-group-flush">
                            <ListGroupItem>Opening Times: 11.00 - 10.30</ListGroupItem>
                            <ListGroupItem>Offers: 25% off with any</ListGroupItem>
                            <ListGroupItem>Contact Number: 0344 800 2400</ListGroupItem>
                        </ListGroup>
                    </Card>
                </Col>
                <Col className="mt-4">

                    <Card className="h-100">
                        <Card.Img variant="top" src="MCD.jpg" style={{ width: "100%" }} />
                        <Card.Body>
                            <Card.Title>KFC</Card.Title>
                            <Card.Text>
                                McDonald's (MCD) is a fast food, limited service restaurant with more than 35,000 restaurants in over 100 countries.
                            </Card.Text>
                        </Card.Body>
                        <ListGroup className="list-group-flush">
                            <ListGroupItem>Opening Times: 8.00 - 1.00</ListGroupItem>
                            <ListGroupItem>Offers: N/A</ListGroupItem>
                            <ListGroupItem>Contact Number: 0344 800 2400</ListGroupItem>
                        </ListGroup>
                    </Card>
                </Col>
                <Col className="mt-4">

                    <Card className="h-100">
                        <Card.Img variant="top" src="bowling.jpg" style={{ width: "100%" }} />
                        <Card.Body>
                            <Card.Title>HollyWood Bowl</Card.Title>
                            <Card.Text>
                                Hollywood Bowl Bracknell is part of The Hollywood Bowl Group; the UKs best ten pin bowling alley packed with bowling fun, arcades, food, drink and activities!
                            </Card.Text>
                        </Card.Body>
                        <ListGroup className="list-group-flush">
                            <ListGroupItem>Opening Times: 11.00 - 12.30</ListGroupItem>
                            <ListGroupItem>Offers: N/A</ListGroupItem>
                            <ListGroupItem>Contact Number: 0344 800 2400</ListGroupItem>
                        </ListGroup>
                    </Card>
                </Col>
            </Row>
        </Container >

    );
}

export default Places;