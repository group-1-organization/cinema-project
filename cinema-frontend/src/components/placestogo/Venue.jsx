import Col from "react-bootstrap/Col"
import Card from "react-bootstrap/Card"
import ListGroup from "react-bootstrap/ListGroup"
import ListGroupItem from "react-bootstrap/ListGroupItem"




const Venue = ({ venue }) => {

    return (
        <Col className="mt-4">
            <Card className="h-100" bg="dark">
                <Card.Img variant="top" src={`/images/${venue.picture}`} alt={venue.picture} style={{ width: "100%" }} />
                <Card.Body>
                    <Card.Title>{venue.name}</Card.Title>
                    <Card.Text>
                        {venue.description}
                    </Card.Text>
                </Card.Body>
                <ListGroup className="list-group-flush">
                    <ListGroupItem className="bg-dark">Opening Times: {venue.opening}</ListGroupItem>
                    <ListGroupItem className="bg-dark">Offers: {venue.offer}</ListGroupItem>
                    <ListGroupItem className="bg-dark">Contact Number: {venue.contact}</ListGroupItem>
                </ListGroup>
            </Card>
        </Col>

    )
}

export default Venue;