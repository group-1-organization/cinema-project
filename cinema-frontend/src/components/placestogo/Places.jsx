import Container from "react-bootstrap/Container"
import Row from "react-bootstrap/Row"
import Jumbotron from "react-bootstrap/Jumbotron"
import Button from "react-bootstrap/Button"
import { useEffect, useState } from "react"
import axios from "axios"
import { Link } from 'react-router-dom';

import Venue from "./Venue"



const Places = () => {
    const [venue, setVenue] = useState([]);

    useEffect(() => {
        getVenues();
    }, []);

    const getVenues = () => {
        axios.get("http://localhost:5000/cinema/venue").then((response) => {
            setVenue(response.data);
        }).catch((error) => {
            console.log(error);
        })
    };



    return (
        <div>
            <div className="page-header">
                <h1>Places to go</h1>
            </div>
            <br />

            <Container>

                <center>

                    <h6>
                        Here is just a few of the great local venues!
                    </h6>
                    <p>
                        <Link to={"/gettingthere"}><Button className="button-colour" >How to find us</Button></Link>
                    </p>
                </center>


                <Row xs={1} lg={3}>
                    {venue.map((venue) => (
                        <Venue venue={venue} />
                    ))}
                </Row>
            </Container >
        </div>

    );
}

export default Places;