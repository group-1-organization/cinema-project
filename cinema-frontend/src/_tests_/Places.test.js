import renderer, { act } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Places from '../components/placestogo/Places';
import Venue from '../components/placestogo/Venue';

test('Places page matches snapshot', () => {
    const tree = renderer.create(
        <Router>
            <Places />
        </Router>);

    expect(tree).toMatchSnapshot();
});

test('Venue matches snapshot', () => {

    const venue = {
        _id: "60ec4b6d7b36692e24ee1f55",
        name: "KFC",
        picture: "KFC.jpg",
        opening: "9.00 - 10.30",
        description: "KFC is an American fast food restaurant chain headquartered in Louisville, Kentucky that specializes in fried chicken",
        contact: "0344 800 2400",
        offer: "N/A"
    }
    const tree = renderer.create(<Venue venue={venue} />);

    expect(tree).toMatchSnapshot();
});