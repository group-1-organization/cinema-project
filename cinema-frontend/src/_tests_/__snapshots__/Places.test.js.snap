// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Places page matches snapshot 1`] = `
<div>
  <div
    className="page-header"
  >
    <h1>
      Places to go
    </h1>
  </div>
  <br />
  <div
    className="container"
  >
    <center>
      <h6>
        Here is just a few of the great local venues!
      </h6>
      <p>
        <a
          href="/gettingthere"
          onClick={[Function]}
        >
          <button
            className="button-colour btn btn-primary"
            disabled={false}
            type="button"
          >
            How to find us
          </button>
        </a>
      </p>
    </center>
    <div
      className="row row-cols-lg-3 row-cols-1"
    />
  </div>
</div>
`;

exports[`Venue matches snapshot 1`] = `
<div
  className="mt-4 col"
>
  <div
    className="h-100 card bg-dark"
  >
    <img
      className="card-img-top"
      src="/images/KFC.jpg"
      style={
        Object {
          "width": "100%",
        }
      }
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        KFC
      </div>
      <p
        className="card-text"
      >
        KFC is an American fast food restaurant chain headquartered in Louisville, Kentucky that specializes in fried chicken
      </p>
    </div>
    <div
      className="list-group-flush list-group"
      onKeyDown={[Function]}
    >
      <div
        className="bg-dark list-group-item"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        Opening Times: 
        9.00 - 10.30
      </div>
      <div
        className="bg-dark list-group-item"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        Offers: 
        N/A
      </div>
      <div
        className="bg-dark list-group-item"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        Contact Number: 
        0344 800 2400
      </div>
    </div>
  </div>
</div>
`;
